### PR TITLE
RDM-1929: AboutToSubmit callback for Divorce (Solicitors)

### PR DESCRIPTION
**Pull request by Daniel-Lam**
# Description

Added `/sanitise` callback endpoint for allowing case data to be transformed, by concatenating selected D8 fields, as per https://tools.hmcts.net/jira/browse/RDM-1929.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Unit tests of the `SanitiseService` class in `SanitiseServiceTest`.

**Test Configuration**:

* Hardware:
* O/S and version: Windows 7 Enterprise
* JDK: 1.8

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/SanitiseService.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                caseData.getD8PetitionerFirstName(),
+                BLANK_SPACE,
+                caseData.getD8PetitionerLastName());
+        caseData.setD8DerivedPetitionerCurrentFullName(isNotBlank(petitionerFullName) ? petitionerFullName : null);
+
+        String petitionerCorrespondenceAddress = Objects.nonNull(caseData.getD8PetitionerCorrespondenceAddress())
+                ? deriveAddress(caseData.getD8PetitionerCorrespondenceAddress()) : null;
+        caseData.setD8DerivedPetitionerCorrespondenceAddress(petitionerCorrespondenceAddress);
+
+        String respondentName = StringUtils.join(
```
Comment (mthava) [inline src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/SanitiseService.java]: Could have moved name derivation into a separate method. The same logic has been repeated few times in this method.

In fact, all the concatenation code can be moved to a single method, with a signature,

String combine(char separator, String... valuesToCombine){
}

Comment (Daniel-Lam) [reply src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/SanitiseService.java]: Good point - looking at refactoring this now

File: src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/SanitiseService.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                address.getAddressLine3(),
+                address.getPostTown(),
+                address.getCounty(),
+                address.getPostCode(),
+                address.getCountry()
+        ).filter(StringUtils::isNotBlank)
+                .collect(Collectors.joining(LINE_SEPARATOR));
+    }
+
+    private String combine(String separator, String... valuesToCombine) {
```
Comment (mthava) [inline src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/SanitiseService.java]: could have used the logic in the deriveAddressMethod - same method could be used for both names and address

Comment (Daniel-Lam) [reply src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/SanitiseService.java]: Very true - will modify accordingly

**Pull request closed without merging**